packages/gluestick/src/generator/generator.js

```diff
@@ -0,0 +1,76 @@
+/* @flow */
+
+import type {
+  Logger,
+  Generator,
+  WrittenTemplate,
+  GeneratorOptions,
+} from '../types';
+
+const path = require('path');
+
+const { isValidEntryPoint } = require('../generator/utils');
+const requireGenerator = require('./requireGenerator');
+const parseConfig = require('./parseConfig');
+const writeTemplate = require('./writeTemplate');
+
+type Command = {
+  generatorName: string;
+  entityName: string;
+  options?: GeneratorOptions;
+}
+
+/**
+ * Starts generator routine.
+ *
+ * @param command {Object} { generatorName, entityName, entryPoint, options } Command object
+ * @param logger {Object} logger Logger instance
+ */
+module.exports = (command: Command, logger: Logger): void => {
+  const {
+    generatorName,
+    entityName,
+    options = {},
+  } = command;
+  const {
+    entryPoint = '',
+  } = options;
+
+  const predefinedGenerators = [
+    'component',
+    'reducer',
+    'container',
+  ];
+
+  if (predefinedGenerators.find((element) => element === generatorName)) {
+    if (!isValidEntryPoint(entryPoint, logger)) {
+      return;
+    }
+  }
+
+  if (!/^[a-zA-Z0-9/_-]+$/.test(entityName)) {
+    logger.error('Invalid name specified');
+    return;
```

Comment:
I think we should not kill the process (see https://www.nczonline.net/blog/2014/02/04/maintainable-node-js-javascript-avoid-process-exit/) but instead throw errors. 

In this case though, it is not like a code error but a error message (using their `logger.error`) to the user when typing the command. I think in this case it's better this way although I am not sure 100% sure.